yb-voyager/src/pgss/parser.go

```diff
@@ -0,0 +1,241 @@
+			mapping.MinExecTime = header
+		case "max_time", "max_exec_time":
+			mapping.MaxExecTime = header
+		case "stddev_time", "stddev_exec_time":
+			mapping.StddevExecTime = header
+		}
+	}
+
+	return mapping
+}
+
+// Validate validates the column mapping by checking each field directly
+func (m *ColumnMapping) Validate() error {
+	var missingFields []string
+	if m.QueryID == "" {
+		missingFields = append(missingFields, "queryid")
+	}
+	if m.Query == "" {
+		missingFields = append(missingFields, "query")
+	}
+	if m.Calls == "" {
+		missingFields = append(missingFields, "calls")
+	}
+	if m.Rows == "" {
+		missingFields = append(missingFields, "rows")
+	}
+	if m.TotalExecTime == "" {
+		missingFields = append(missingFields, "total_exec_time")
+	}
+	if m.MeanExecTime == "" {
+		missingFields = append(missingFields, "mean_exec_time")
+	}
+	if m.MinExecTime == "" {
+		missingFields = append(missingFields, "min_exec_time")
+	}
+	if m.MaxExecTime == "" {
+		missingFields = append(missingFields, "max_exec_time")
+	}
+	if m.StddevExecTime == "" {
+		missingFields = append(missingFields, "stddev_exec_time")
+	}
+
+	if len(missingFields) > 0 {
+		return fmt.Errorf("missing required fields in CSV headers: %v", missingFields)
+	}
+	return nil
+}
+
+// ParseFromCSV parses a CSV file and returns normalized QueryStats entries
+func ParseFromCSV(csvPath string) ([]QueryStats, error) {
+	file, err := os.Open(csvPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open PGSS CSV file %s: %w", csvPath, err)
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+
+	headers, err := reader.Read()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CSV headers: %w", err)
+	}
+
+	mapping := CreateColumnMapping(headers)
+	if err = mapping.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid PGSS CSV structure: %w", err)
+	}
+
+	var entries []QueryStats
+	lineNumber := 1 // Header is line 0
+	validEntries := 0
+	skippedEntries := 0
+	for {
+		record, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, fmt.Errorf("failed to read CSV row at line %d: %w", lineNumber, err)
+		} else if len(headers) != len(record) {
+			return nil, fmt.Errorf("invalid PGSS CSV structure: headers count does not match record count at line %d", lineNumber)
+		}
+
+		entry, err := parseCSVRecord(headers, record, mapping)
+		if err != nil {
+			log.Warnf("Skipping invalid PGSS record at line %d: %v", lineNumber, err)
+			skippedEntries++
+			continue
+		}
+
+		// Entry is valid if parsing succeeded
+		entries = append(entries, *entry)
+		validEntries++
+		lineNumber++
+	}
+
+	log.Infof("PGSS CSV parsing completed: %d valid entries, %d skipped entries", validEntries, skippedEntries)
+	return entries, nil
+}
+
+// parseCSVRecord converts a single CSV record to QueryStats using the column mapping
+func parseCSVRecord(headers []string, record []string, mapping *ColumnMapping) (*QueryStats, error) {
```

Comment:
I'm a little confused about why we need the `ColumnMapping` structure. 

When i think about the logic, it would go: 
1. read headers.
2. read each record
    a. validate that record len == header len
    b. Initialize QueryStats struct.
    c. for each column in record: 
          i. switch case on header(col): 
                parse into right variable and populate the QueryStats struct's field. 
          ii. Do any validations to ensure that there are no missing fields. 

For cases where there are multiple possible names for a column (total_time, total_exec_time), the switch case can group the column names together.